src/main/java/service/impl/StorageStrategyProcessor.java

```diff
@@ -0,0 +1,39 @@
+package service.impl;
+
+import db.Storage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.StrategyProcessor;
+import strategy.OperationBalance;
+import strategy.OperationPurchase;
+import strategy.OperationReturn;
+import strategy.OperationStrategy;
+import strategy.OperationSupply;
+
+public class StorageStrategyProcessor implements StrategyProcessor {
+
+    private final Storage storage;
+    private final Map<FruitTransaction.Operation, OperationStrategy> operationStrategyMap;
+
+    public StorageStrategyProcessor(Storage storage) {
```

Comment:
It is better to get the Map as the parameter of the Controller than hard code here
```suggestion
    public StorageStrategyProcessor(Storage storage, Map<FruitTransaction.Operation, OperationStrategy> operationStrategyMap) {
```